library/manageiq_provider.py

```diff
@@ -296,35 +343,47 @@ def add_or_update_provider(self, provider_name, provider_type, endpoints, zone,
 
         Returns:
             the added or updated provider id, whether or not a change took
-            place and a short message describing the operation executed
+            place and a short message describing the operation executed,
+            including the authentication validation status
         """
-        message = ""
         zone_id = self.find_zone_by_name(zone or 'default')
         # check if provider with the same name already exists
         provider_id = self.find_provider_by_name(provider_name)
-        updates = {}
         if provider_id:  # provider exists
             updates = self.required_updates(provider_id, endpoints, zone_id, provider_region)
-            if updates:
-                self.update_provider(provider_id, provider_name, endpoints, zone_id, provider_region)
-                message = "Successfuly updated {provider} provider".format(
-                    provider=provider_name)
-            else:
-                message = "Provider %s already exists" % provider_name
-            return dict(
-                provider_id=provider_id,
-                changed=self.changed,
-                msg=message,
-                updates=updates
-            )
+            if not updates:
+                return dict(changed=self.changed,
+                            msg="Provider %s already exists" % provider_name)
+
+            old_validation_details = self.auths_validation_details(provider_id)
+            operation = "update"
+            self.update_provider(provider_id, provider_name, endpoints, zone_id, provider_region)
+            roles_with_changes = set(updates["Added"]) | set(updates["Updated"])
         else:  # provider doesn't exists, adding it to manageiq
+            updates = None
+            old_validation_details = {}
+            operation = "add"
             provider_id = self.add_new_provider(provider_name, provider_type,
                                                 endpoints, zone_id, provider_region)
-            return dict(
-                provider_id=provider_id,
-                changed=self.changed,
-                msg="Successfuly added %s provider" % provider_name
-            )
+            roles_with_changes = [e['endpoint']['role'] for e in endpoints]
+
+        authtypes_to_verify = []
+        for e in endpoints:
+            if e['endpoint']['role'] in roles_with_changes:
+                authtypes_to_verify.append(e['authentication']['authtype'])
+        success, details = self.verify_authenticaion_validation(provider_id, old_validation_details, authtypes_to_verify)
+        operations = {"add": {"past": "added", "present": "addition"},
+                      "update": {"past": "updated", "present": "update"}}
```

Comment:
Nice, you fixed "updateed" :+1: 
Consider simpler
```
operation, operation_past = "add", "added"
...
operation, operation_past = "update", "update"
```
Or maybe use only "addition"/"update" with "Successful {operation} of {provider} provider." wording.